### DISCUSSION TITLE
Is there a known issue with `send_keys()` and running tests in parallel?

### DISCUSSION THREAD

**Discussion by tking16** [appium/appium#20087] [Q&A] [answered]
I have a few tests which involve signing with dynamically created accounts named `atsuite+randomchars@gmail.com`, I've noticed running the tests in parallel the string sequence is either missing some characters or putting them in the complete wrong place. This is using:
- v4.0.0 of Appium Python Client 
- Appium 2.5.4
- xcuitest-driver 7.15.2
![Screenshot 2024-05-10 at 17 57 52](https://github.com/appium/appium/assets/13576824/3c535917-fba5-4990-9fc1-ab43344bcc73 "Failing test screenshot")
![Screenshot 2024-05-10 at 17 58 54](https://github.com/appium/appium/assets/13576824/63e4028b-544c-45c6-99c1-2df532440210 "Another failing test, this time it missed the 'm' in gmail")

Running on Simulators, works fine with a singular device, but when Selenium Grid is introduced I'm getting errors 
```
    def type_email(self, email_string):
        email = self._driver.wait_for(self.locators.email_field)
        email.click()
        email.send_keys(email_string)
```

**Comment by tking16** [comment]
I've out found there's a cap called `maxTypingFrequency` gonna experiment with that by halving the default

**Comment by tking16** [comment]
Setting maxTypingFrequency to 30 did nothing 😔 Might I add that this is all in headless mode

**Comment by tking16** [comment]
Nothing I try is working, @mykola-mokhnach @KazuCocoa @jlipps QQ: Do the capabilities `interKeyDelay` and `sendKeyStrategy` still work for Appium 2?

**Comment by jlipps** [comment, marked as answer]
If the problem only shows up when running in parallel, and you've checked the Appium logs to ensure that the correct string for send keys makes it to the Appium server, then the issue is probably not with Appium. Maybe running multiple simulators on the same machine is too taxing and the simulators are not behaving properly.

**Reply by tking16** [reply]
Agh silly me, I was setting the capabilities in the wrong place (the test run command rather than the Appium start command) which is why nothing was changing, seeing less errors now, hopefully can get rid of them as I play with the number
